language/tools/move-cli/src/main.rs

```diff
@@ -163,6 +170,25 @@ impl Move {
         Ok(src_dirs)
     }
 
+    /// Same as `get_compilation_deps`, but do a best-effort filtering to exclude dep files already present in 
+    /// `to_compile`. This prevents "duplicate definition" warnings from the compiler
+    fn get_filtered_deps(&self, to_compile: &[String]) -> Result<Vec<String>> {
+        fn get_stem(s: &String) -> Option<&str> {
+            Path::new(s).file_stem().map(|stem| stem.to_str()).flatten()
+        }
+        let deps = self.get_compilation_deps()?;
+        let to_compile_filenames = move_lang::find_move_filenames(to_compile, true)?;   
+        // convert /path/to/File.move to File.move
+        let file_names: Vec<&str> = to_compile_filenames.iter().map(|s| get_stem(s)).flatten().collect();
+        // exclude any filenames in `deps` that do not match File.move
+        // Note: this is best-effort in at least two ways:
+        // (1) It will wrongly filter modules in the same name under different addresses (e.g., 0x1/M.move and 0x2/M.move)
+        // (2) It will fail to filter modules with the same filename, but different module names (e.g., 0x1/M in both M1.move and M2.move)
```

Comment:
(3) It may cause modules to be lost? (e.g., 0x1/M and 0x1/N in the deps' `M.move` but only 0x1/M in the updated `M.move` (although I guess this should not be an issue for stdlib modules).